docs/source/notes/distributed_autograd.rst

```diff
@@ -295,6 +295,87 @@ you can refer to **Distributed Autograd Algorithm Smart mode** section in the
 
 Distributed Optimizer
 ^^^^^^^^^^^^^^^^^^^^^
-Coming soon...
+
+The :class:`~torch.distributed.optim.DistributedOptimizer` operates as follows:
+
+1. Takes a list of remote parameters (:class:`~torch.distributed.rpc.RRef`) to
+   optimize. These could also be local parameters wrapped within a local
+   ``RRef``.
+2. Takes a :class:`~torch.optim.Optimizer` class as the local
+   optimizer to run on all distinct ``RRef`` owners.
+3. The distributed optimizer creates an instance of the local ``Optimizer`` on
+   each of the worker nodes and holds an ``RRef`` to them.
+4. When :meth:`torch.distributed.optim.DistributedOptimizer.step` is invoked,
+   the distributed optimizer uses RPC to remotely execute all the local
+   optimizers on the appropriate remote workers.
+5. If multiple concurrent distributed optimizers are updating the same
+   parameters on a worker, these updates are serialized via a lock.
+
+Simple end to end example
```

Comment:
Other section names use first capital letter

Reply:
This is true for top level sections but not necessarily other sections. I mostly capitalized based on whether it looked appropriate :)